text/0005-disallow-whitespace-in-entityuid.md

```diff
@@ -0,0 +1,94 @@
+3. Customers are anticipated to build meta-permissions layers that restrict callers to manipulating policy store contents for only certain namespaces. This may lead to policies such as `forbid(...) when {context.namespace = "ExampleCo::Photoflash"};`. There is a risk that an unauthorized actor could bypass this restriction by using a namespace with embedded spaces. 
+
+In addition to the above, pentesters and malicious actors will explore ways to craft Cedar policy statements that look like they do one thing, but actually do another. The existence of little known parsing behaviors provide an avenue for exploration by this audience, as demonstrated by the second example in the opening of this RFC:
+
+```
+permit( 
+  principal == ExampleCo::Photoflash::User:://"alice"
+
+  "bob", //TODO: Try various tricks to obfuscate this line from readers
+  action,
+  resource
+);
+```
+
+## Detailed design
+The UID will be treated as a single identifier with special structure (rather than tokenizing it into `::` separated sequence of identifiers).
+
+## Drawbacks
+This is a breaking change. We do not generally want breaking changes in Cedar. In the worst case scenario, a customer may have an existing `forbid` statement that relies on this behavior as illustrated below. By changing the behavior, the `forbid` statement will error at runtime and be skipped, and hence the system could fail open.
+
+```
+forbid(
+  principal in ExampleCo:: Photoflash::UserGroup::"someGroup", //Accidential typo, but valid syntax
+  action == ExampleCo:: Photoflash::Action::"write",
+  resource
+);
+```
+
+This breaking change is only being considered due to the relative newness of Cedar and the probability of breaking a real production scenario being low, and it can only be done if the community agrees that altering this behavior will pay long term dividends that outweight the risk of near-term impacts.
+
+## Alternatives
+
+We could change the behavior of `EntityTypeName::from_str()` without changing the syntax used in Cedar policies themselves. In `EntityTypeName::from_str()`, whitespace would be disallowed throughout -- either before, after, or in the middle of the typename. In Cedar policies, whitespace would still be allowed in every position it’s allowed today.  This would avoid a breaking change to the Cedar language and avoid diverging from the behavior of most mainstream programming languages, which do allow whitespace around namespaces and namespace separators.  This alternative would fully address the Examples 1, 2, and 3 in this PDF, and while it would not address the malicious policy example given, that should be weighed against the impact of a breaking change to the Cedar language and diverging from other commonly-used languages.
```

Comment:
The alternative seems incompatible with the proposal summary where it says we should disallow whitespaces and comments in `EntityUID`, which consists of an entity type and a uid.